modules/unsupported/cql-json/src/test/java/org/geotools/filter/text/cqljson/CQLJsonParsingTest.java

```diff
@@ -0,0 +1,598 @@
+package org.geotools.filter.text.cqljson;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import org.geotools.filter.FilterFactoryImpl;
+import org.geotools.filter.text.cql2.CQLException;
+import org.junit.Assert;
+import org.junit.Test;
+import org.opengis.filter.Filter;
+
+public class CQLJsonParsingTest {
+    @Test
+    public void convertJsonToPredicates() throws JsonProcessingException, CQLException {
+        String swimmingpool =
+                "{"
+                        + "             \"and\": ["
+                        + "                {"
+                        + "                   \"eq\": {"
+                        + "                      \"property\": \"swimming_pool\","
+                        + "                      \"value\": true"
+                        + "                   }"
+                        + "                },"
+                        + "                {"
+                        + "                   \"or\": ["
+                        + "                      {"
+                        + "                         \"gt\": {"
+                        + "                            \"property\": \"floor\","
+                        + "                            \"value\": 5"
+                        + "                         }"
+                        + "                      },"
+                        + "                      {"
+                        + "                         \"like\": {"
+                        + "                            \"property\": \"material\","
+                        + "                            \"value\": \"brick%\""
+                        + "                         }"
+                        + "                      },"
+                        + "                      {"
+                        + "                         \"like\": {"
+                        + "                            \"property\": \"material\","
+                        + "                            \"value\": \"%brick\""
+                        + "                         }"
+                        + "                      } "
+                        + "                   ]"
+                        + "                }"
+                        + "             ]"
+                        + "          }";
+
+        CQLJsonCompiler cqlJsonCompiler =
+                new CQLJsonCompiler(swimmingpool, new FilterFactoryImpl());
+        cqlJsonCompiler.compileFilter();
+        Filter filter = cqlJsonCompiler.getFilter();
+        Assert.assertEquals(
+                "[[ swimming_pool = true ] AND [[ floor > 5 ] OR [ material is like brick% ] OR [ material is like %brick ]]]",
+                filter.toString());
```

Comment:
I strongly recommend using ` ECQL.toCQL(filter)` rather than `filter.toString()` to have a stable string representation to compare against. As a bonus it should also show the difference between `5` and '"5"`.

Reply:
Good point.  I changed the test to use ECQL.toCQL rather than toString for stability.